### PR TITLE
Build against TF2.6.0

### PR DESCRIPTION
**Pull request by seanpmorgan**
# Description

Testing next TF release candidate

### CODE REVIEW

**Comment by bhack** [comment]
@googlebot I consent.

**Comment by bhack** [comment]
I've fixed one API change but we need to check what happens now with `layer_test`. 
The more general issue is that as we depend on internal/private testing API we are fragile:

https://github.com/tensorflow/tensorflow/issues/28601
https://github.com/tensorflow/tensorflow/issues/34408

**Comment by bhack** [comment]
I made a little bit of wheels bisecting on a single sample test and It seems that the last working TF version was:

`pip install tf-nightly-cpu==2.6.0.dev20210615`

The next "working" wheel but with failing test is:

`pip install tf-nightly-cpu==2.6.0.dev20210618`

**Comment by bhack** [comment]
Ok I've fixed all the errors with a few workarounds.
We need to check with Keras If the import tricks are caused by some repo sync issues with Keras standalone or something else.

More in general we need to find a final solution for:
https://github.com/tensorflow/tensorflow/issues/28601
/cc @yarri-oss

File: tensorflow_addons/utils/test_utils.py [hunk 1]
```diff
@@ -26,8 +26,10 @@
 from tensorflow_addons.utils import resource_loader
 
 # TODO: copy the layer_test implementation in Addons.
-from tensorflow.python.keras.testing_utils import layer_test  # noqa: F401
-
+if tf.__version__[:3] > "2.5":
+    from keras.testing_utils import layer_test  # noqa: F401
+else:
+    from tensorflow.python.keras.testing_utils import layer_test  # noqa: F401
```
Comment (bhack) [inline tensorflow_addons/utils/test_utils.py:32]: @qlzh727 As we are near to the release do we sill need to maintain this conditional import?

File: tensorflow_addons/utils/types.py [hunk 2]
```diff
@@ -20,7 +20,10 @@
 import tensorflow as tf
 
 # TODO: Remove once https://github.com/tensorflow/tensorflow/issues/44613 is resolved
-from tensorflow.python.keras.engine import keras_tensor
+if tf.__version__[:3] > "2.5":
+    from keras.engine import keras_tensor
+else:
+    from tensorflow.python.keras.engine import keras_tensor
```
Comment (bhack) [inline tensorflow_addons/utils/types.py:26]: @qlzh727 Ditto

File: tensorflow_addons/utils/test_utils.py (hunk 1 above)
Comment (qlzh727) [reply tensorflow_addons/utils/test_utils.py:32]: yea, please keep those conditional imports here. We will check with API owner about exposing certain API to internal name space.

File: tensorflow_addons/utils/types.py (hunk 2 above)
Comment (qlzh727) [reply tensorflow_addons/utils/types.py:26]: Same.

**Comment by qlzh727** [comment]
Are `keras.testing_utils.layer_test` and `keras.engine.keras_tensor` the only two module Addon are importing at the moment?

**Comment by bhack** [comment]
> Are `keras.testing_utils.layer_test` and `keras.engine.keras_tensor` the only two module Addon are importing at the moment?

From the tests that was failing it seems like this. If you can give a pass over https://github.com/tensorflow/tensorflow/issues/28601

**Comment by bhack** [comment]
So It Is almost ready.
@seanpmorgan Do we have any deprecation to add or any API to remove?

**Comment by bhack** [comment]
@fsx950223 Do we have any PR to merge before the release?

**Comment by fsx950223** [comment]
LGTM

**Comment by seanpmorgan** [comment]
> So It Is almost ready.
> @seanpmorgan Do we have any deprecation to add or any API to remove?

I don't believe so. Which components required increasing the minimum version to 2.4? No issue with the increase but just want to document what it was for

**Comment by bhack** [comment]
You could check with all the last changes if It will go to build again with 2.3 but evaluating the currently available human resources here I will suggest to not going to support more then 3 releases.

**Comment by AakashKumarNain** [comment]
> You could check with all the last changes if It will go to build again with 2.3 but evaluating the currently available human resources here I will suggest to not going to support more then 3 releases.

Agree with @bhack 3 releases sounds good to me

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -262,7 +262,7 @@ If you need a custom C++/Cuda op for your test, compile your ops with
 
 ```bash
 python configure.py
-pip install tensorflow==2.5.0 -e ./ -r tools/install_deps/pytest.txt
+pip install tensorflow==2.6.0rc0 -e ./ -r tools/install_deps/pytest.txt
```
Comment (lgeiger) [inline CONTRIBUTING.md:265]:
```suggestion
pip install tensorflow==2.6.0 -e ./ -r tools/install_deps/pytest.txt
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -290,7 +290,7 @@ docker run --gpus all --rm -it -v ${PWD}:/addons -w /addons gcr.io/tensorflow-te
 
 Configure:
 ```bash
-python3 -m pip install tensorflow==2.5.0
+python3 -m pip install tensorflow==2.6.0rc0
```
Comment (lgeiger) [inline CONTRIBUTING.md:293]:
```suggestion
python3 -m pip install tensorflow==2.6.0
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -329,7 +329,7 @@ quickly, as Bazel has great support for caching and distributed testing.
 To test with Bazel:
 
 ```bash
-python3 -m pip install tensorflow==2.5.0
+python3 -m pip install tensorflow==2.6.0rc0
```
Comment (lgeiger) [inline CONTRIBUTING.md:332]:
```suggestion
python3 -m pip install tensorflow==2.6.0
```

File: tensorflow_addons/callbacks/average_model_checkpoint.py [hunk 6]
```diff
@@ -78,20 +78,20 @@ def set_model(self, model):
                 "with MovingAverage or StochasticAverage"
             )
 
-    def _save_model(self, epoch, logs):
+    def _save_model(self, **kwargs):
```
Comment (lgeiger) [inline tensorflow_addons/callbacks/average_model_checkpoint.py:81]: Should we change this to the following in order to also support positional arguments:
```suggestion
    def _save_model(self, *args, **kwargs):
```
This seems to be more closely match the [signature used in the keras code](https://github.com/keras-team/keras/blob/83ac1ed7f8293c1921996a7ddd61fe7e8e66c5ca/keras/callbacks.py#L1395).

File: tensorflow_addons/callbacks/average_model_checkpoint.py [hunk 7]
```diff
@@ -78,20 +78,20 @@ def set_model(self, model):
...
         optimizer = self._get_optimizer()
         assert isinstance(optimizer, AveragedOptimizerWrapper)
 
         if self.update_weights:
             optimizer.assign_average_vars(self.model.variables)
-            return super()._save_model(epoch, logs)
+            return super()._save_model(**kwargs)
```
Comment (lgeiger) [inline tensorflow_addons/callbacks/average_model_checkpoint.py:87]:
```suggestion
            return super()._save_model(*args, **kwargs)
```

File: tensorflow_addons/callbacks/average_model_checkpoint.py [hunk 8]
```diff
@@ -78,20 +78,20 @@ def set_model(self, model):
...
         else:
             # Note: `model.get_weights()` gives us the weights (non-ref)
             # whereas `model.variables` returns references to the variables.
             non_avg_weights = self.model.get_weights()
             optimizer.assign_average_vars(self.model.variables)
             # result is currently None, since `super._save_model` doesn't
             # return anything, but this may change in the future.
-            result = super()._save_model(epoch, logs)
+            result = super()._save_model(**kwargs)
```
Comment (lgeiger) [inline tensorflow_addons/callbacks/average_model_checkpoint.py:95]:
```suggestion
            result = super()._save_model(*args, **kwargs)
```

**Review by lgeiger — DISMISSED** [review]
Thanks for upgrading to TF 2.6.0, I just have two minor comments from the sideline. Other than that looking forward to the new release.

**Comment by seanpmorgan** [comment]
LGTM. @bhack if you approve I can get this merged and shipped

**Review by lgeiger — APPROVED** [review]

**Review by bhack — APPROVED** [review]

**Comment by bhack** [comment]
Thank you all.. also if am not in the release notes greetings. 😜

**Comment by seanpmorgan** [comment]
> Thank you all.. also if am not in the release notes greetings. 😜

Ah it's auto generated from release drafter. Looks like combined authors may not have been added. Thx as always @bhack!

**Comment by bhack** [comment]
> Looks like combined authors may not have been added.

Yes it Is not supported.

**Comment by keshavbhandari** [comment]
Tensorflow addons does not support Tensorflow 2.7.0 now. Colab has updated its TF version to 2.7.0.

**Comment by seanpmorgan** [comment]
> Tensorflow addons does not support Tensorflow 2.7.0 now. Colab has updated its TF version to 2.7.0.

TFA 0.15 was just released. You can pip install it in colab... or if you'd like to raise an issue on colab to update TFA that'd be great. I've tried in the past to mixed results.

**Comment by bhack** [comment]
@seanpmorgan Can you make a short post in the forum in the announcement category?

**Comment by seanpmorgan** [comment]
https://discuss.tensorflow.org/t/tfa-0-15-released/5725

**Pull request merged**